### DISCUSSION TITLE
Supid question: Get State of LED device

### DISCUSSION THREAD

**Discussion by sorted01** [awawa-dev/HyperHDR#225] [Q&A]
Hi awawa,

it's me again :)
I have one question about json-rpc within Home Assistant.
For HDR this works fine with the value template:

```
command_on: 'curl -X POST -i "http://192.168.178.27:8090/json-rpc" -d ''{"command":"videomodehdr","HDR":1}'''
command_off: 'curl -X POST -i "http://192.168.178.27:8090/json-rpc" -d ''{"command":"videomodehdr","HDR":0}'''
command_state: "curl -k http://192.168.178.27:8090/json-rpc -H 'Content-Type: application/json' -d '{\"command\":\"serverinfo\"}'"
value_template: >
  {% if value_json.info.videomodehdr == 1 %}
  {{ true }}
  {% else %}
  {{ false }}
  {% endif %}
```

Now i want to do the same for the LED device

```
command_on: 'curl -X POST -i "http://192.168.178.27:8090/json-rpc" -d ''{"command":"componentstate","componentstate":{"component":"LEDDEVICE","state": true}}'''
command_off: 'curl -X POST -i "http://192.168.178.27:8090/json-rpc" -d ''{"command":"componentstate","componentstate":{"component":"LEDDEVICE","state": false}}'''
command_state: "curl -k http://192.168.178.27:8090/json-rpc -H 'Content-Type: application/json' -d '{\"command\":\"serverinfo\"}'"
value_template: >
```

Can you tell me how the command_state and the value template should look like? I don't get it working... :/

I know there is the integration available. But once i use the LED switch of the integration it's not staring to use the information of the screen capturing (PicCap with WLED) but showing a solid color with a new source from Home Assistant. If I activate the LED device with the json-rpc it works like it should. But currently I just don't know how to get the status :D

Thank you!

Best Regards
sorteds

**Comment by awawa-dev** [comment]
Hi
Not a stupid question and it's in fact more complicated. HDR tone mapping is global & special state and similar solution is impossible for LED device. You have an array in info.components where one of the elements is LEDDEVICE and you can read its state. But here is even harder part: it only describes current instance state. The other instances could have LEDDEVICE in different state, so you should also iterate them using switchTo ...
So usually to turn off/on LEDSTRIP globally rather command similar to following should be use and that state should be checked instead:
```command_off: 'curl -X POST -i "http://192.168.178.27:8090/json-rpc" -d ''{"command":"componentstate","componentstate":{"component":"ALL","state": false}}'''```

**Reply by sorted01** [reply]
And once more thank you for the fast reply. I will give it a try later this evening and let you know if I succeeded :D
